pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -563,13 +563,33 @@ var pgBuiltins = map[string]builtinDefinition{
 				{"encoding_id", types.Int},
 			},
 			ReturnType: tree.FixedReturnType(types.String),
+			Fn:         getEncodingNameForEncodingID,
+			Info:       notUsableInfo,
+		},
+	),
+
+	"getdatabaseencoding": makeBuiltin(
+		tree.FunctionProperties{Category: categorySystemInfo},
+		tree.Overload{
+			Types:      tree.ArgTypes{},
+			ReturnType: tree.FixedReturnType(types.String),
 			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
-				if args[0].Compare(ctx, DatEncodingUTFId) == 0 {
-					return datEncodingUTF8ShortName, nil
+				row, err := ctx.InternalExecutor.QueryRow(
+					ctx.Ctx(), "getdatabaseencoding",
+					ctx.Txn,
+					"SELECT encoding FROM pg_database WHERE datname = $1", ctx.SessionData.Database)
```

Comment:
sorry bit slow to realise this - looks like we're always returning UTF-8 for this anyway. We populate `pg_database` with a virtual schema, and we always return the same col for this rowtype:
 
https://github.com/cockroachdb/cockroach/blob/bb14400cdd1397f47c3781b4830199f169cef497/pkg/sql/pg_catalog.go#L1084

Perhaps we should also just return UTF-8 here as well. Make a comment on the virtual schema that if we ever change the value of that, we should change it here and vice versa.

Sorry for the round-aboutness with getting this out there.

Reply:
(note: i would generally try avoid calling another query inside here unless you had to, which led me down this path to finding this :] ideally, if it's configurable, we'd have a function that lets us figure this out without shelling out to another query)

Reply:
So, rather than using InternalExecutor to execute the query you are recommending me to 
just return UTF-8 and add a comment to virtual schema to that if we change value there we update it here too.

like use in ```pg_client_encoding``` in ```pg_builtins.go```